File: app/src/main/java/com/monk/rxjava2/imetate2/Observer.java
package com.monk.rxjava2.imetate2;

/**
 * @author monk
 * @date 2019-03-01
 */
public interface Observer<T> {
    void onSubscribe(Disposable disposable);

    void onNext(T t);

    void onCompleted();

    void onError(Throwable throwable);
}

File: module_fragment/src/main/java/com/monk/modulefragment/rxjava2/imetate2/Observer.kt
package com.monk.modulefragment.rxjava2.imetate2

/**
 * @author monk
 * @date 2019-03-01
 */
interface Observer<T> {
    fun onSubscribe(disposable: Disposable?)
    fun onNext(t: T)
    fun onCompleted()
    fun onError(throwable: Throwable?)
}